gpytorch/kernels/grid_kernel.py

```diff
@@ -34,10 +34,16 @@ def __init__(self, base_kernel, grid, interpolation_mode=False, active_dims=None
         super(GridKernel, self).__init__(active_dims=active_dims)
         self.interpolation_mode = interpolation_mode
         self.base_kernel = base_kernel
-        self.register_buffer("grid", grid)
+        self.register_buffer_list('grid', grid)
         if not self.interpolation_mode:
             self.register_buffer("full_grid", create_data_from_grid(grid))
 
+    def register_buffer_list(self, base_name, tensors):
+        """Helper to register several buffers at once under a single base name"""
+        self.__setattr__(base_name, tensors)
```

Comment:
This sets `self.grid` to point to `tensors`. But `register_buffer` in L45 makes copies of each tensor, so this pointer no longer points to the right tensors. One work around is remove this `setattr` and instead make a property for called `grid` that returns a list `[self.grid_0,...]`.
e.g.

```
@property
    def grid(self):
        return [getattr(self, "grid" + '_' + str(i)) for i in range(self.num_dims)]
```
Note you'll need to add a `num_dims` attribute

Reply:
Ah, thanks for pointing this out, I didn't realize that `register_buffer` makes copies of tensors. Will fix.